Dockerfile

```diff
@@ -25,17 +29,33 @@ WORKDIR /app
 COPY . /app
 RUN bundle exec rake install
 
+FROM ruby:3.0-alpine
 
-FROM ruby:2.7-alpine
+ARG UID=${UID:-1000}
+ARG GID=${GID:-1000}
 
 COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
 
-RUN apk add \
-    bash \
-    git \
-    libc6-compat
-
-VOLUME /wiki
 WORKDIR /wiki
+RUN apk add --update \
+            --no-cache \
+            bash \
+            git \
+            shadow \
+            curl \
+            libc6-compat \
+    && rm -rf /var/cache/apk/* \
+    && groupmod -g $GID www-data \
+    && adduser -u $UID -S www-data -G www-data
+
 COPY docker-run.sh /docker-run.sh
+RUN chmod +x /docker-run.sh
+USER www-data
+VOLUME /wiki
+EXPOSE 4567
+
+HEALTHCHECK --start-period=2s --interval=5m --timeout=3s \
+  CMD curl -f http://localhost:4567 || exit 1
+
 ENTRYPOINT ["/docker-run.sh"]
+CMD ["/bin/sh", "/docker-run.sh"]
```

Comment:
What is the motivation for this change?

Reply:
The main purpose of CMD is to provide defaults for an executable run, in which case it specifies in the ENTRYPOINT. If desired we can remove it.